### PR TITLE
[cleanup] remove unexpected `await`s

### PR DESCRIPTION
**Pull request by bkendall**
### Description

Removing the `await-thenable` warning and fixing all the errors.

### Scenarios Tested

- [ ] give a few more commands some sanity testing

### CODE REVIEW

File: src/emulator/functionsEmulator.ts [hunk 1]
```diff
@@ -424,7 +424,7 @@ export class FunctionsEmulator implements EmulatorInstance {
               added = await this.addPubsubTrigger(this.args.projectId, definition);
               break;
             case Constants.SERVICE_AUTH:
-              added = await this.addAuthTrigger(this.args.projectId, definition);
+              added = this.addAuthTrigger(this.args.projectId, definition);
```
Comment (samtstern) [inline src/emulator/functionsEmulator.ts:427]: This one is a bit suspicious to me.  I guess because auth is in-process it can be added synchronously?

File: src/hosting/functionsProxy.ts [hunk 2]
```diff
@@ -49,6 +49,6 @@ export default function(
       }
     }
 
-    return await proxyRequestHandler(url, `${destLabel} Function ${rewrite.function}`);
+    return Promise.resolve(proxyRequestHandler(url, `${destLabel} Function ${rewrite.function}`));
```
Comment (samtstern) [inline src/hosting/functionsProxy.ts:52]: Why this change?

**Review by samtstern — APPROVED** [review]
Two small comments

File: src/emulator/functionsEmulator.ts (hunk 1 above)
Comment (bkendall) [reply src/emulator/functionsEmulator.ts:427]: yup - it's a sync function, not async, so no need to await

File: src/hosting/functionsProxy.ts (hunk 2 above)
Comment (bkendall) [reply src/hosting/functionsProxy.ts:52]: Two fold:
1. proxyRequestHandler is _not_ async, and does _not_ return a promise (async and await are removed)
2. Since the signature of the function required a promise (and the JSDocs too), the returned handler must be wrapped in a promise.

File: src/hosting/functionsProxy.ts [hunk 3]
```diff
@@ -25,7 +25,7 @@ export interface FunctionProxyRewrite {
 export default function(
   options: FunctionsProxyOptions
 ): (r: FunctionProxyRewrite) => Promise<RequestHandler> {
-  return async (rewrite: FunctionProxyRewrite) => {
+  return (rewrite: FunctionProxyRewrite) => {
```
Comment (yuchenshi) [inline src/hosting/functionsProxy.ts:28]: nit: You may want to wrap the whole function body around `new Promise` to keep the behavior the same. Before, any errors in the function body will be returned as a rejected Promise. Now it just synchronously throws (which can be really bad), unless you wrap everything.

**Review by yuchenshi — APPROVED** [review]
LGTM (with nit above)

File: src/hosting/functionsProxy.ts (hunk 3 above)
Comment (bkendall) [reply src/hosting/functionsProxy.ts:28]: Makes sense. I'll wrap it. Thanks.
